src/developer-quickstart.md

```diff
@@ -0,0 +1,482 @@
+- **Sway**: the domain-specific language crafted for the FuelVM; it is inspired by Rust.
+- **Forc**: the build system and package manager for Sway, similar to Cargo for Rust.
+
+## Understand Sway Program Types
+
+There are four types of Sway programs:
+
+- `contract`
+- `predicate`
+- `script`
+- `library`
+
+Contracts, predicates, and scripts can produce artifacts usable on the blockchain, while a library is simply a project designed for code reuse and is not directly deployable.
+
+The main features of a smart contract that differentiate it from scripts or predicates are that it is callable and stateful.
+
+A script is runnable bytecode on the chain which can call contracts to perform some task. It does not represent ownership of any resources and it cannot be called by a contract.
+
+|             | deployable on the blockchain:       | can have state:      | callable on the blockchain: | designed for code reuse: |
+|-------------|-------------------------------------|----------------------|-----------------------------|--------------------------|
+| contract  | ✅                                  | ✅                   | ✅                          | ❌                        |
+| predicate | ✅                                  | ❌                    | ❌                           | ❌                        |
+| script    | ❌                                   | ❌                    | ❌                           | ❌                        |
+| library   | ✅ (via a contract or predicate) | ❌ | ❌                           | ✅                       |
+
+See [the chapter on program types](../sway-program-types/index.md) for more information.
+
+## Your First Sway Project
+
+We'll build a simple counter contract with two functions: one to increment the counter, and one to return the value of the counter.
+
+A few pieces of info that will be helpful before moving on:
+
+- This guide was created using VSCode as the code editor.
+- Download the Sway language extension in VSCode to get syntax highlighting, keyboard shortcuts, and more.
+- Download the rust-analyzer extension in VSCode to get syntax highlighting, code completion, and more.
+
+**Start by creating a new, empty folder. We'll call it `fuel-project`.**
+
+### Writing the Contract
+
+First, let's [install the Sway toolchain](https://github.com/FuelLabs/fuelup).
+
+Then with `forc` installed, create a contract project inside of your `fuel-project` folder:
+
+```sh
+cd fuel-project
+forc new counter_contract
+```
+
+Here is the project that `Forc` has initialized:
+
+```console
+├── fuel-project
+    ├── Cargo.toml
+    ├── Forc.toml
+    ├── src
+    │   └── main.sw
+    └── tests
+        └── harness.rs
+```
+
+`Forc.toml` is the _manifest file_ (similar to `Cargo.toml` for Cargo or `package.json` for Node) and defines project metadata such as the project name and dependencies. You'll notice a `cargo.toml` file because `forc` uses `cargo` under the hood.
```

Comment:
```suggestion
`Forc.toml` is the _manifest file_ (similar to `Cargo.toml` for Cargo or `package.json` for Node) and defines project metadata such as the project name and dependencies. You'll notice a `Cargo.toml` file because `forc` uses `cargo` under the hood.
```